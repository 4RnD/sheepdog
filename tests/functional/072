#!/bin/bash

# Test restart with missing object in stale directory

seq=`basename $0`
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1        # failure is the default!

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

_cleanup

MD=false

for i in 0 1; do
    _start_sheep $i
done
_wait_for_sheep 2

_cluster_format -c 1

_start_sheep 2
_wait_for_sheep 3

$COLLIE vdi create test 8M -P
$COLLIE cluster shutdown

mv $STORE/0/obj/007c2b2500000001 $STORE/0/obj/.stale/007c2b2500000001.1
mv $STORE/1/obj/807c2b2500000000 $STORE/1/obj/.stale/807c2b2500000000.1

for i in 0 1 2; do
    _start_sheep $i
done

_wait_for_sheep 3
_wait_for_sheep_recovery 0

for i in 0 1 2; do
    $COLLIE vdi list -p $((7000+$i)) | _filter_short_date
done

$COLLIE vdi check test
