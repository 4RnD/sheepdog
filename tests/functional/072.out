QA output created by 072
using backend plain store
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0  8.0 MB  8.0 MB  0.0 MB DATE   7c2b25      1                22
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0  8.0 MB  8.0 MB  0.0 MB DATE   7c2b25      1                22
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0  8.0 MB  8.0 MB  0.0 MB DATE   7c2b25      1                22
finish check&repair test
